### DISCUSSION TITLE
How do I setup alchemy provider when using `configureChains` with multiple networks

### DISCUSSION THREAD

**Discussion by kelokchan** [wevm/wagmi#597] [Q&A]
Currently our dApp supports ethereum and polygon network, and I need the wagmi client to support both of them. When I use `configureChains`, I can specify the `alchemyProvider({ alchemyId })` as the `providers` array. However in Alchemy, every network has its own separate app hence I will have multiple API key. How should I pass in the keys to the `alchemyProvider`?

**Comment by tmm** [comment]
You should be able to use the same `alchemyId` across multiple chains so you only need to provide it once.

**Comment by kelokchan** [comment]
![image](https://user-images.githubusercontent.com/5347725/176627987-736d15c1-bd44-4ae2-b589-37e268732be2.png)
We have 6 apps in alchemy and each has their own api key, how should I configure it then?

**Reply by tmm** [reply]
Even though the Alchemy interface says they are for a specific chain, a single key works across many chains so just popping one of those into `alchemyProvider` should be fine.

If you still want to use individual apps, you can set up a custom [`jsonRpcProvider`](https://wagmi.sh/docs/providers/jsonRpc).

**Reply by 0xCardiE** [reply]
Why do they then say its just for one chain and make you choose each time? Is there any merit to what you choose that makes difference?

**Reply by tmm** [reply]
Not sure. The [Alchemy SDK docs](https://docs.alchemy.com/reference/alchemy-sdk-quickstart#2-after-installing-the-app-you-can-then-import-and-use-the-sdk) don't mention that API keys are tied to a specific network either so I'm guessing it's just a UX thing in the dashboard.
